### PR TITLE
feat: Foreign calls compiling and basic print executed in nargo

### PR DESCRIPTION
**Pull request by vezenovm**
# Description

If you run nargo execute --experimental-ssa and change the Prover.toml in the brillig_oracle test you will see the println being used

## Summary\*

This PR sets out to

### Example

Before:

```

```

After:

```

```

## Documentation

- [ ] This PR requires documentation updates when merged.

  

  - [ ] I will submit a noir-lang/docs PR.

  

  - [ ] I will request for and support Dev Rel's help in documenting this PR.

  
  

## Additional Context

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

File: crates/nargo_cli/tests/test_data_ssa_refactor/brillig_oracle/src/main.nr [hunk 1]
```diff
@@ -6,7 +6,7 @@ fn main(x: Field)  {
 
 
 #[oracle(oracle_print_impl)]
-unconstrained fn oracle_print(x : Field) -> Field {}
+unconstrained fn oracle_print(_x : Field) {}
```
Comment (ludamad) [inline crates/nargo_cli/tests/test_data_ssa_refactor/brillig_oracle/src/main.nr:9]: Right, I was going to look into making unused not warn, but this makes sense

**Review by ludamad — APPROVED** [review]
Neat, LGTM so far

File: crates/nargo/src/ops/execute.rs [hunk 2]
```diff
@@ -11,9 +13,39 @@ pub fn execute_circuit(
 ) -> Result<WitnessMap, NargoError> {
     let mut blocks = Blocks::default();
     let solver_status = solve(backend, &mut initial_witness, &mut blocks, circuit.opcodes)?;
-    if matches!(solver_status, PartialWitnessGeneratorStatus::RequiresOracleData { .. }) {
-        todo!("Add oracle support to nargo execute")
-        // TODO 1557
+
+    // TODO: Nargo only supports "oracle_print_impl" functions that print a singular value and nothing else
```
Comment (kevaundray) [inline crates/nargo/src/ops/execute.rs:17]: link to issue

File: crates/noirc_evaluator/src/brillig/brillig_gen.rs [hunk 3]
```diff
@@ -273,7 +290,7 @@ impl BrilligGen {
         brillig.convert_ssa_function(func);
 
         brillig.push_code(BrilligOpcode::Stop);
-
+        dbg!(brillig.obj.byte_code.clone());
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/brillig/brillig_gen.rs:293]:
```suggestion
```

File: crates/nargo/src/ops/execute.rs (hunk 2 above)
Comment (kevaundray) [reply crates/nargo/src/ops/execute.rs:17]: This merges into Adam's branch so we can ignore and do cleanup there

Comment (vezenovm) [reply crates/nargo/src/ops/execute.rs]: done

File: crates/noirc_evaluator/src/brillig/brillig_gen.rs [hunk 4]
```diff
@@ -184,6 +186,21 @@ impl BrilligGen {
...
+                let result_ids = dfg.instruction_results(instruction_id);
+
+                let input_registers =
+                    vecmap(arguments, |value_id| self.convert_ssa_value(*value_id, dfg));
+                let output_registers =
+                    vecmap(result_ids, |value_id| self.convert_ssa_value(*value_id, dfg));
+
+                let opcode = BrilligOpcode::ForeignCall {
+                    function: func.to_owned(),
+                    destination: RegisterValueOrArray::RegisterIndex(output_registers[0]),
```
Comment (kevaundray) [inline crates/noirc_evaluator/src/brillig/brillig_gen.rs:199]: Noting that this does not work for functions with more than one return value and input

**Pull request merged**